tidy3d/plugins/adjoint/components/medium.py

```diff
@@ -190,14 +192,18 @@ def store_vjp(
 
 @register_pytree_node_class
 class JaxCustomMedium(CustomMedium, JaxObject):
-    """A :class:`.Medium` registered with jax."""
+    """A :class:`.CustomMedium` registered with ``jax``.
+    Note: The gradient calculation assumes uniform field across the pixel.
+    Therefore, the accuracy degrades as the pixel size becomes large
+    with respect to the field variation.
+    """
```

Comment:
So just to clarify something here:
* Whatever operations used in the vjp maker for CustomMedium do not need to be backpropped. They are simply for evaluating the integral, which is why we can directly use xarray methods on the field data. 
* When determining the vjp, we need to perform a volume integral, which is currently approximating by interpolation at the position of the data point. However, to do this properly, we need to take into account that the CustomMedium has interpolation options of `nearest` and `linear`. 
    * if `nearest`, each dataset has "influence" over its own rectangular domain, so we can determine the vjp by just performing a volume integral over that domain.
    * if `linear`, the domains of each neighboring pixel are overlapping and we'd need to add a spatially-dependent weight this volume integral. I considered disallowing `linear` in `JaxCustomMedium` but tried it with the current point-wise approximation and it worked fine, so leaving it allowed for now.
* One of my next plans (for 1.9rc3 or 1.10) is to do the volume integral for `nearest`, which should handle the z-constant medium properly if field variation along that dimension is large. 
* Any back propagation through `interp` would need to be defined in `JaxDataArray` but it's a bit complicated so I haven't considered that yet. I figure users could implement their own interpolation within their custom functions using `sel`. The thing is that I'm not 100% sure how xarray / scipy handle interpolation, especially in dimensions > 2 but I also haven't looked into it too much yet due to other priorities. Could be nice to have, but doesn't affect the VJP I think.

Reply:
side note: I think the way to define `JaxDataArray.interp(**kwargs)` would be to define a `kwarg` and `self.coords` -dependent mask containing the interpolation weights (in the same shape as `self.values`) and then return the result of `jnp.sum(weights * self.values)`. Jax should be able to handle the rest. I just haven't looked into how to construct those weights.

Reply:
Yeah, that makes sense, when I said "back-propping through the sel/interpolate" I was talking exactly about what's happening in CustomMedium on the forward pass depending on `'nearest'` or `'linear'`.

Reply:
Yeah that's how it works, `legume` has a `np.interp` vjp, I think I had the impression that it can be somewhat slow and or memory intensive though, probably even more so for >1D. https://github.com/fancompute/legume/blob/cfa800e8ffabde8f5d621a7d38a77c59a1132780/legume/primitives.py#L226-L250

Reply:
Oh well avoiding an explicit `for` loop may help if it's slow...

Reply:
yeah why not